spikewrap/data_classes/base.py

```diff
@@ -48,14 +47,18 @@ class will accept `Dict[str, Union[str, List[str]]]` and
                 ), "Run names must be string or list of strings"
                 self.sessions_and_runs[key] = [value]
 
-    def preprocessing_sessions_and_runs(self):  # TODO: type hint
-        """"""
-        ordered_ses_names = list(
-            chain(*[[ses] * len(runs) for ses, runs in self.sessions_and_runs.items()])
-        )
-        ordered_run_names = list(
-            chain(*[runs for runs in self.sessions_and_runs.values()])
-        )
+    def flat_sessions_and_runs(self) -> List[Tuple[str, str]]:
+        """
+        This returns the sessions and runs dictionary flattened so that
+        sessions and runs can be iterated over conveniently. `ordered_run_names`
+        is flattened to a long list of all runs, while `ordered_ses_name` carries
+        the corresponding session for each run.
+        """
+        ordered_run_names = []
+        ordered_ses_names = []
+        for ses, run in self.sessions_and_runs.items():
+            ordered_run_names += run
+            ordered_ses_names += [ses] * len(run)
 
         return list(zip(ordered_ses_names, ordered_run_names))
```

Comment:
I see why you do this, you find more convenient to have prepared a list of tuples rather than using a double for loop multiple times.
I guss the double for loop looks like
```python
for ses_name, run_names in self.preprocessing_sessions_and_runs():
    for run_name in run_names:
        utils.update(self.data, ses_name, run_name, {"0-raw": None})
        utils.update(self.sync, ses_name, run_name, None)
```
I can also suggest another implementation for this method:
```python
return [   
    (ses, run)
    for ses, runs in self.preprocessing_sessions_and_runs()
    for run in runs
]
```
Also a generator could be considered.